src/Quizzer.API.DAL/QuizzerDbContext.cs

```diff
@@ -47,16 +47,16 @@ public QuizzerDbContext(DbContextOptions<QuizzerDbContext> options)
                 .HasMany<UserAnswerEntity>()
                 .WithOne(userAnswer => userAnswer.Question)
                 .OnDelete(DeleteBehavior.Cascade);
-            
+
             modelBuilder.Entity<AnswerEntity>()
                 .HasMany<SelectedAnswerEntity>()
                 .WithOne(selectedAnswer => selectedAnswer.Answer)
-                .OnDelete(DeleteBehavior.Cascade);
-            
+                .OnDelete(DeleteBehavior.Restrict);
+
             modelBuilder.Entity<UserAnswerEntity>()
                 .HasMany<SelectedAnswerEntity>()
                 .WithOne(selectedAnswer => selectedAnswer.UserAnswer)
-                .OnDelete(DeleteBehavior.Cascade);
+                .OnDelete(DeleteBehavior.Restrict);
```

Comment:
So what does this changed rules do with our ER model? Still the Selected answer will be deleted whether user or user answer will be deleted?

Reply:
I think it won't let you. If selected answers exist you have to remove them first.

Reply:
It's not dependant on User, only Answer and UserAnswer

Reply:
> It's not dependant on User, only Answer and UserAnswer

Yeah i meant UserAnswer but wrote it separately.

Reply:
> I think it won't let you. If selected answers exist you have to remove them first.

@KoubaFromPucks there should be done tests to verify this kind of behavior. And maybe then alter the facades because this is important information.

Reply:
So you want to make unit test on DBContext or repository or some integration tests?

Reply:
@KoubaFromPucks Once this merges, try to make a test where you create a userAnswer and answer with at lest one SelectedAnswer and try to delete them. I think you do not have to save these tests for this but just let me know and I will edit the facades.

Reply:
ok